### PR TITLE
contrib/emicklei/go-restful/v3: add integration for newest version

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### What does this PR do?

Adds support for the [latest version of `go-restful`](https://github.com/emicklei/go-restful#using-go-modules).

### Motivation

The integration for previous versions can't be used with `v3`.

### Reviewer's Checklist

- [x] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

**Review by rarguelloF — CHANGES_REQUESTED** [review]
@michaelbeaumont 👋 Thanks for your contribution!
Component name needs to be updated, everything else LGTM.

File: contrib/emicklei/go-restful/v3/restful.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
+
+	"github.com/emicklei/go-restful/v3"
+)
+
+const componentName = "emicklei/go-restful"
```
[the remark this answers is not in the document]
Comment (michaelbeaumont) [reply contrib/emicklei/go-restful/v3/restful.go]: done!

**Comment by rarguelloF** [comment]
@michaelbeaumont there are some linter errors, could you take a look? 🙏 
https://github.com/DataDog/dd-trace-go/actions/runs/5420976820/jobs/9856100211?pr=2088

**Comment by michaelbeaumont** [comment]
> @michaelbeaumont there are some linter errors, could you take a look? 🙏 https://github.com/DataDog/dd-trace-go/actions/runs/5420976820/jobs/9856100211?pr=2088

Done

**Review by rarguelloF — APPROVED** [review]

**Review by katiehockman — APPROVED** [review]

**Pull request merged**